api/renterd/api.go

```diff
@@ -188,6 +200,144 @@ func (s *server) syncerConnectHandler(w http.ResponseWriter, req *http.Request,
 	}
 }
 
+func (s *server) contractScanHandler(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
+	var cbr ContractScanRequest
+	if err := json.NewDecoder(req.Body).Decode(&cbr); err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	cm := (s.cm).(*chain.Manager)
+	session, err := renter.NewSession(cbr.NetAddress, cbr.HostKey, s.w, s.tp, cm)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	settings, err := session.ScanSettings()
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	api.WriteJSON(w, settings)
+}
+
+func (s *server) contractFormHandler(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
+	var cfr ContractFormRequest
+	if err := json.NewDecoder(req.Body).Decode(&cfr); err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	cm := (s.cm).(*chain.Manager)
+	session, err := renter.NewSession(cfr.NetAddress, cfr.HostKey, s.w, s.tp, cm)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
```

Comment:
we'll want to add a `defer session.Close()` here

Reply:
As far as I can tell Session does not have a Close method